### ISSUE TITLE
"Manually pumped" executor, for use e.g. inside another event loop?

### ISSUE THREAD

**Issue by lithiumfluorine** [tzcnt/TooManyCooks#149]
Hello! Let's say I'm writing a game, and I want to use the default CPU executor for performing various background work, but also want to be able to schedule work (or co_await myself) back onto my dedicated main thread from the workers, something approximately like this (not super-familiar with this library but I hope it gets the idea across):

```cpp
ManuallyProcessedExecutor mainExecutor; // <- new thing?

tmc::task<void> loadEntity(...) {
    co_await readFile(...); // could be a network request or anything else that's awaitable really
    parseFile(...); // some CPU-bound work to be done on the worker
    co_await tmc::resume_on(mainExecutor);
    spawnEntity(...); // must be done on main thread for sync reasons
}

int main() {
    setup();
    while (true) {
        mainExecutor.process_all_pending_work();
        update();
        // Something inside update occasionally calls:
        // tmc::spawn(loadEntity(...)).run_on(tmc::cpu_executor()).detach();
        render();
    }
}
```

Essentially, I'm "stuck" in another event loop that I need to keep running (so I can't "just" use a `tmc::ex_cpu_st`, I think?), but still want to process incoming work posted via tmc on a regular basis. I assume I would need to implement a custom executor for this? Thank you.

**Comment by lithiumfluorine** [comment]
Oh that was quick! Gonna try it out as soon as I have the time. Thanks ❤️
Yeah, feel free!

**Comment by tzcnt** [comment]
I've updated the documentation with an example based on your sample. I've made a few tweaks:
- using `post()` instead of `spawn().run_on().detach()`, since we are posting from outside of an executor
- demonstrating how to handle both tasks that are unimportant (can complete on a future frame) and tasks that are important (we need them to complete this frame before we can proceed)

**Comment by tzcnt** [comment]
Released in v1.3.0
